samples/java_springboot/21.corebot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/MainDialog.java

```diff
@@ -227,7 +227,7 @@ public MainDialog(FlightBookingRecognizer withLuisRecognizer,
             );
             Activity message = MessageFactory
                 .text(messageText, messageText, InputHints.IGNORING_INPUT);
-            stepResult = stepContext.getContext().sendActivity(message).thenApply(sendResult -> null);
+            return stepContext.getContext().sendActivity(message).thenApply(sendResult -> null);
```

Comment:
This is not ok, the way that it was before is ok, as the result will be added in the stepResult and then once the sendActivity is done, it will execute the replaceDialog of L236.

We can rollback this